base/linalg/bunchkaufman.jl

```diff
@@ -150,37 +150,42 @@ function A_ldiv_B!{T<:BlasComplex}(B::BunchKaufman{T}, R::StridedVecOrMat{T})
     end
 end
 
-function det(F::BunchKaufman)
-    if F.info > 0
-        return zero(eltype(F))
-    end
-
+function logabsdet(F::BunchKaufman)
     M = F.LD
     p = F.ipiv
     n = size(F.LD, 1)
 
+    if F.info > 0
+        return eltype(F)(-Inf), zero(eltype(F))
+    end
+    s = one(real(eltype(F)))
     i = 1
-    d = one(eltype(F))
+    abs_det = zero(real(eltype(F)))
     while i <= n
         if p[i] > 0
-            # 1x1 pivot case
-            d *= M[i,i]
+            elm = M[i,i]
+            s *= sign(elm)
+            abs_det += log(abs(elm))
             i += 1
         else
             # 2x2 pivot case. Make sure not to square before the subtraction by scaling
             # with the off-diagonal element. This is safe because the off diagonal is
             # always large for 2x2 pivots.
             if F.uplo == 'U'
-                d *= M[i, i + 1]*(M[i,i]/M[i, i + 1]*M[i + 1, i + 1] -
+                elm = M[i, i + 1]*(M[i,i]/M[i, i + 1]*M[i + 1, i + 1] -
                     (issymmetric(F) ? M[i, i + 1] : conj(M[i, i + 1])))
+                s *= sign(elm)
+                abs_det += log(abs(elm))
             else
-                d *= M[i + 1,i]*(M[i, i]/M[i + 1, i]*M[i + 1, i + 1] -
+                elm = M[i + 1,i]*(M[i, i]/M[i + 1, i]*M[i + 1, i + 1] -
                     (issymmetric(F) ? M[i + 1, i] : conj(M[i + 1, i])))
```

Comment:
I guess it is trying to avoid overflow?  Anyway, this stuff is probably not performance critical because it is linear-time; the expensive part in practice is computing the factorization.  So let's leave it as-is.